include/deal.II/grid/grid_tools.h

```diff
@@ -1079,6 +1079,48 @@ namespace GridTools
     std::map<typename Triangulation<Container::dimension,Container::space_dimension>::active_cell_iterator,
     typename Container::active_cell_iterator> &patch_to_global_tria_map);
 
+  /**
+   * This function runs through the degrees of freedom defined by the
+   * Container and for each dof constructs a vector of active_cell_iterators
+   * representing the cells of support of the nodal basis element
+   * at that degree of freedom. This function was designed for the
+   * implementation of local projections, for instance the Clement interpolant,
+   * in conjunction with other local patch functions like
+   * GridTools::build_triangulation_from_patch.
+   *
+   * Containers built on top of Triangulation<dim> or
+   * parallel:distributed::Triangulation<dim> are supported and handled
+   * appropriately.
+   *
+   * It is necessary that the finite element underlying the MeshType used has
+   * degrees of freedom that are logically associated to a vertex, line, quad,
+   * or hex.  This includes both nodal degrees of freedom and other modal types
+   * of dofs that are associated to an edge, etc.  The result is the patch of
+   * cells representing the support of the basis element associated to the
+   * degree of freedom.  For instance using an FE_Q finite element, we obtain
+   * the standard patch of cells touching the degree of freedom and then others
+   * that take care of possible hanging node constraints.  Using a FE_DGQ finite
+   * element, the degrees of freedom are logically considered to be "interior" to
+   * the cells so the patch would be the cell on which the degree of freedom is
+   * located.
+   *
+   * @tparam MeshType The MeshType should be a DoFHandler<dim> or hp::DoFHandler<dim>.
```

Comment:
Omit the template arguments to `DoFHandler` and `hp::DoFHandler` -- they don't add information, and one could also call this function for a codimension-1 or -2 mesh.